### DISCUSSION TITLE
How to configure debug？

### DISCUSSION THREAD

**Discussion by Anhod** [oxigraph/oxigraph#468] [Q&A]
Hello, congratulations on the release of v0.3.14. I'm learning the source code of oxigraph and want to understand the workflow of inserting and querying data, but I find it boring and inefficient to just read against the source code, so I want to debug it to advance my work. The compiler I am using is VS Code, and I am not sure how to configure the launch.json file because I don't know much about it, so I would like to ask for your help, thank you very much!

**Comment by Tpt** [comment]
Hi! Thank you! That's a great question. I am not very familiar with VSCode (I am a CLion user). I believe the best entry point is using the debugger on the command line endpoint (for example "cargo run load --location my_storage_file --file my_file.nt" for the loading and  "cargo run query --location my_storage_file --query 'my query'" for querying, both evaluated in `./store`).

There is also some technical documents [in the wiki](https://github.com/oxigraph/oxigraph/wiki) you might find interesting.
